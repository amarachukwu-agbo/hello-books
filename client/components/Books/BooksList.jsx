import React from 'react';
import propTypes from 'prop-types';
import Book from './Book.jsx';

/**
 * @description stateless component for rendering books
 *
 * @param {object} props - properties of the component
 * @returns {Node} - react node containing the pagination Component
 */

const BooksList = (props) => {
  const { books } = props;
  if (!books.length) {
    return (
      <div className="row center">
        <h3 className="red-text flow-text">
          Oops! Seems no books are available at the moment </h3>
      </div>
    );
  }

  return (
    <div className="row">
      {
        books.map((book, index) =>
          <Book key={index} book={book} />)
      }
    </div>
  );
};

// Prop type validation
BooksList.propTypes = {
  books: propTypes.array.isRequired,
};

export default BooksList;
