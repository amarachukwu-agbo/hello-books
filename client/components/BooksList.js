import React from 'react';
import Book from './Book';

const BooksList = (props) => {
  const { books } = props;
  if (!books.length) {
    return (
        <div className="row center">
            <h3 className="red-text flow-text">Oops! Seems no books are available at the moment </h3>
        </div>
    );
  }

  return (
        <div className="row book-list">
            {
                books.map((book, index) =>
                    <Book key= { index } book= { book } />)
            }
        </div>
  );
};

export default BooksList;
