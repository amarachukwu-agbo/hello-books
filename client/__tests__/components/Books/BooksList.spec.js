import React from 'react';
import { shallow } from 'enzyme';
import BooksList from '../../../components/Books/BooksList';

const props = {
  books: [
    {
      id: 94,
      title: 'Macbeth',
      author: 'A.J.Hartley',
      description: 'Macbeth: A Novel.',
      subject: 'Romance',
      imageURL: 'https://images/mages/I/51l6MwpodJL._AA300_.jpg',
      quantity: 70,
      borrowCount: 0,
      favCount: 1,
      upvotes: 0,
      downvotes: 0,
      createdAt: '2018-05-10T03:11:52.181Z',
      updatedAt: '2018-07-06T00:58:12.625Z',
      bookReviews: [],
    },
    {
      id: 95,
      title: 'Born a Crime',
      author: 'Trevor Noah',
      description: 'Born a crime',
      subject: 'Biography',
      imageURL: 'https://image/images/I/61mJhMWAq8L._AA300_.jpg',
      quantity: 18,
      borrowCount: 3,
      favCount: 1,
      upvotes: 1,
      downvotes: 0,
      createdAt: '2018-05-10T03:11:52.181Z',
      updatedAt: '2018-07-10T08:49:01.366Z',
      bookReviews: [],
    },
  ],
};
describe('<BooksList />', () => {
  it('should render a list of books correctly', () => {
    const tree = shallow(<BooksList {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render <h3> when no books are available', () => {
    const books = [];
    const wrapper = shallow(<BooksList books = { books } />);
    expect(wrapper.find('h3').text())
      .toEqual('Oops! Seems no books are available at the moment');
  });
});
