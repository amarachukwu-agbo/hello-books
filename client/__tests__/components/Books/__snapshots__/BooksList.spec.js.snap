// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BooksList /> should render a list of books correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BooksList
    books={
      Array [
        Object {
          "author": "A.J.Hartley",
          "bookReviews": Array [],
          "borrowCount": 0,
          "createdAt": "2018-05-10T03:11:52.181Z",
          "description": "Macbeth: A Novel.",
          "downvotes": 0,
          "favCount": 1,
          "id": 94,
          "imageURL": "https://images/mages/I/51l6MwpodJL._AA300_.jpg",
          "quantity": 70,
          "subject": "Romance",
          "title": "Macbeth",
          "updatedAt": "2018-07-06T00:58:12.625Z",
          "upvotes": 0,
        },
        Object {
          "author": "Trevor Noah",
          "bookReviews": Array [],
          "borrowCount": 3,
          "createdAt": "2018-05-10T03:11:52.181Z",
          "description": "Born a crime",
          "downvotes": 0,
          "favCount": 1,
          "id": 95,
          "imageURL": "https://image/images/I/61mJhMWAq8L._AA300_.jpg",
          "quantity": 18,
          "subject": "Biography",
          "title": "Born a Crime",
          "updatedAt": "2018-07-10T08:49:01.366Z",
          "upvotes": 1,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Book
          book={
            Object {
              "author": "A.J.Hartley",
              "bookReviews": Array [],
              "borrowCount": 0,
              "createdAt": "2018-05-10T03:11:52.181Z",
              "description": "Macbeth: A Novel.",
              "downvotes": 0,
              "favCount": 1,
              "id": 94,
              "imageURL": "https://images/mages/I/51l6MwpodJL._AA300_.jpg",
              "quantity": 70,
              "subject": "Romance",
              "title": "Macbeth",
              "updatedAt": "2018-07-06T00:58:12.625Z",
              "upvotes": 0,
            }
          }
        />,
        <Book
          book={
            Object {
              "author": "Trevor Noah",
              "bookReviews": Array [],
              "borrowCount": 3,
              "createdAt": "2018-05-10T03:11:52.181Z",
              "description": "Born a crime",
              "downvotes": 0,
              "favCount": 1,
              "id": 95,
              "imageURL": "https://image/images/I/61mJhMWAq8L._AA300_.jpg",
              "quantity": 18,
              "subject": "Biography",
              "title": "Born a Crime",
              "updatedAt": "2018-07-10T08:49:01.366Z",
              "upvotes": 1,
            }
          }
        />,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "94",
        "nodeType": "function",
        "props": Object {
          "book": Object {
            "author": "A.J.Hartley",
            "bookReviews": Array [],
            "borrowCount": 0,
            "createdAt": "2018-05-10T03:11:52.181Z",
            "description": "Macbeth: A Novel.",
            "downvotes": 0,
            "favCount": 1,
            "id": 94,
            "imageURL": "https://images/mages/I/51l6MwpodJL._AA300_.jpg",
            "quantity": 70,
            "subject": "Romance",
            "title": "Macbeth",
            "updatedAt": "2018-07-06T00:58:12.625Z",
            "upvotes": 0,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "95",
        "nodeType": "function",
        "props": Object {
          "book": Object {
            "author": "Trevor Noah",
            "bookReviews": Array [],
            "borrowCount": 3,
            "createdAt": "2018-05-10T03:11:52.181Z",
            "description": "Born a crime",
            "downvotes": 0,
            "favCount": 1,
            "id": 95,
            "imageURL": "https://image/images/I/61mJhMWAq8L._AA300_.jpg",
            "quantity": 18,
            "subject": "Biography",
            "title": "Born a Crime",
            "updatedAt": "2018-07-10T08:49:01.366Z",
            "upvotes": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Book
            book={
              Object {
                "author": "A.J.Hartley",
                "bookReviews": Array [],
                "borrowCount": 0,
                "createdAt": "2018-05-10T03:11:52.181Z",
                "description": "Macbeth: A Novel.",
                "downvotes": 0,
                "favCount": 1,
                "id": 94,
                "imageURL": "https://images/mages/I/51l6MwpodJL._AA300_.jpg",
                "quantity": 70,
                "subject": "Romance",
                "title": "Macbeth",
                "updatedAt": "2018-07-06T00:58:12.625Z",
                "upvotes": 0,
              }
            }
          />,
          <Book
            book={
              Object {
                "author": "Trevor Noah",
                "bookReviews": Array [],
                "borrowCount": 3,
                "createdAt": "2018-05-10T03:11:52.181Z",
                "description": "Born a crime",
                "downvotes": 0,
                "favCount": 1,
                "id": 95,
                "imageURL": "https://image/images/I/61mJhMWAq8L._AA300_.jpg",
                "quantity": 18,
                "subject": "Biography",
                "title": "Born a Crime",
                "updatedAt": "2018-07-10T08:49:01.366Z",
                "upvotes": 1,
              }
            }
          />,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "94",
          "nodeType": "function",
          "props": Object {
            "book": Object {
              "author": "A.J.Hartley",
              "bookReviews": Array [],
              "borrowCount": 0,
              "createdAt": "2018-05-10T03:11:52.181Z",
              "description": "Macbeth: A Novel.",
              "downvotes": 0,
              "favCount": 1,
              "id": 94,
              "imageURL": "https://images/mages/I/51l6MwpodJL._AA300_.jpg",
              "quantity": 70,
              "subject": "Romance",
              "title": "Macbeth",
              "updatedAt": "2018-07-06T00:58:12.625Z",
              "upvotes": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "95",
          "nodeType": "function",
          "props": Object {
            "book": Object {
              "author": "Trevor Noah",
              "bookReviews": Array [],
              "borrowCount": 3,
              "createdAt": "2018-05-10T03:11:52.181Z",
              "description": "Born a crime",
              "downvotes": 0,
              "favCount": 1,
              "id": 95,
              "imageURL": "https://image/images/I/61mJhMWAq8L._AA300_.jpg",
              "quantity": 18,
              "subject": "Biography",
              "title": "Born a Crime",
              "updatedAt": "2018-07-10T08:49:01.366Z",
              "upvotes": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
